### ISSUE TITLE
Attributes added dynamically

### ISSUE THREAD

**Issue by thipages** [WebReflection/lighterhtml#10]
Hi,
Is it possible to add HTML attributes dynamically?
I wrote two naive tries 2A, 2B leading to js errors (see below)
Thanks

```javascript
import {render, html} from '//unpkg.com/lighterhtml?module';
//
const attributes='style="color:red;"';
const color="color:blue;";

// Case 1:  OK
const div1=(color)=> html`<div style="${color}">Text</div>`
render (document.body,()=> div1(color));

// Case 2A: NOK : Uncaught Error: 1 values instead of 0 <div , >Text</div>
const div2=(attributes)=> html`<div ${attributes}>Text</div>`
render (document.body,()=> div2(attributes));

// Case 2B: NOK : Uncaught Error: 2 values instead of 1 <div , =", ">Text</div>
const div3=(key,value)=> html`<div ${key}="${value}">Text</div>`
render (document.body,()=> div3("style",color));
```

**Comment by WebReflection** [comment]
Nope, same as in hyperHTML.

**Comment by WebReflection** [comment]
Apologies for the quick reply but the TL;DR is that you can't have partial outputs here 'cause you're not dealing with stings, you're dealing with real DOM, strings are just a facade of the (domtagger) library.

This won't ever change, and AFAIK it's the same in lit-html, as well as hyperHTML, due same concept: it's DOM, you can't represent dynamic attributes via DOM/HTML/XML.

Feel free to ask more, if needed, but this won't ever happen in here, unless W3C will make it possible to define inline dynamic attributes.

**Comment by thipages** [comment]
no problem for the quick answer.
I was thinking that while managing the interpolations in tagged templates within `html `(btw where is it in the code?), we could have a try catch that render a tagged template, result of the former parsed.

**Comment by WebReflection** [comment]
nope, it can't be done like that for various reasons, where these are just a few:

  * performance killer
  * no idea what to look for in your interpolation, there is **no HTML parser here** and never will
  * on idea how to update those interpolations next

TL;DR don't bother, we've discussed this already long time ago, it won't happen, this is not handlebar, if you need runtime created HTML please look elsewhere 'cause it won't happen here.

> btw where is it in the code?

it's the [domtagger](https://github.com/WebReflection/domtagger#domtagger) library, based on the fact template literals are uniques, the DOM is created via a single join, and it will never support what you are asking due technical, and logical, constrains

**Comment by thipages** [comment]
thank you
